package-system/lz4/build-archlinux.sh

```diff
@@ -0,0 +1,168 @@
+        pacman -Qi qemu-system-aarch64 > /dev/null 2>&1
+        if [ $? -ne 0 ]
+        then
+            echo ""
+            echo "qemu-system-aarch64 package needs to be installed."
+            echo "Use pacman to install it"
+            exit 1
+        fi
+        echo ""
+        echo "Cross compiling aarch64 on an amd64 machine validated."
+        echo ""
+    fi
+else
+    echo "Building ${TARGET_AARCH} natively."
+fi
+
+# Set the library name prefix to lz4
+LIB_NAME="lz4"
+
+# Setup the docker arguments for the target architecture
+if [ "${TARGET_AARCH}" = "x86_64" ]
+then
+    echo "Processing Docker for x86_64"
+    TARGET_DOCKER_FILE=Dockerfile.x86_64
+    TARGET_DOCKER_PLATFORM_ARG=linux/amd64
+    DOCKER_IMAGE_NAME=${LIB_NAME}_linux_3p
+elif [ "${TARGET_AARCH}" = "aarch64" ]
+then
+    echo "Processing Docker for aarch64"
+    TARGET_DOCKER_FILE=Dockerfile.aarch64
+    TARGET_DOCKER_PLATFORM_ARG=linux/arm64v8
+    DOCKER_IMAGE_NAME=${LIB_NAME}_linux_aarch64_3p
+else
+    echo "Unsupported architecture ${TARGET_AARCH}"
+    exit 1
+fi
+
+# Make sure docker is installed
+DOCKER_VERSION=$(docker --version)
+if [ $? -ne 0 ]
+then
+    echo "Required package docker is not installed"
+    echo "docker can installed using the 'docker' package"
+    exit 1
+fi
+echo "Detected Docker Version $DOCKER_VERSION"
+
+DOCKER_BUILD_SCRIPT=docker_build_${LIB_NAME}_linux.sh
+
+if [ ! -f $DOCKER_BUILD_SCRIPT ]
+then
+    echo "Invalid docker build script ${DOCKER_BUILD_SCRIPT}"
+    exit 1
+fi
+
+# Prepare the docker file and use the temp folder as the context root
+cp -f ${DOCKER_BUILD_SCRIPT} temp/
+
+pushd temp
+
+# Build the Docker Image
+echo "Building the docker build script for ${DOCKER_IMAGE_NAME}"
+docker buildx build --platform ${TARGET_DOCKER_PLATFORM_ARG} --build-arg DOCKER_BUILD_SCRIPT=$DOCKER_BUILD_SCRIPT -f ../${TARGET_DOCKER_FILE} -t ${DOCKER_IMAGE_NAME}:latest .
+if [ $? -ne 0 ]
+then
+    echo "Error occurred creating Docker image ${DOCKER_IMAGE_NAME}:latest."
+    echo "The following command failed:"
+    echo ""
+    echo "docker buildx build --platform ${TARGET_DOCKER_PLATFORM_ARG} --build-arg DOCKER_BUILD_SCRIPT=$DOCKER_BUILD_SCRIPT -f ../${TARGET_DOCKER_FILE} -t ${DOCKER_IMAGE_NAME}:latest ."
+    echo ""
+    exit 1
+fi
+
+
+# Capture the Docker Image ID
+IMAGE_ID=$(docker images -q ${DOCKER_IMAGE_NAME}:latest)
+if [ -z $IMAGE_ID ]
+then
+    echo "Error: Cannot find Image ID for ${DOCKER_IMAGE_NAME}"
+    exit 1
+fi
+
+
+# Run the Docker Image
+echo "Running build script in the docker image"
+docker run --platform ${TARGET_DOCKER_PLATFORM_ARG} --tty ${DOCKER_IMAGE_NAME}:latest /data/workspace/${DOCKER_BUILD_SCRIPT}
+if [ $? -ne 0 ]
+then
+    echo Failed to build from docker image ${DOCKER_IMAGE_NAME}:latest
+    echo "To log into and troubleshoot the docker container, run the following command:"
+    echo ""
+    echo "docker run --platform ${TARGET_DOCKER_PLATFORM_ARG} --tty ${DOCKER_IMAGE_NAME}:latest"
```

Comment:
This will need `-it` to run interactively, or else you will log into the shell but will get no response
```suggestion
    echo "docker run --platform ${TARGET_DOCKER_PLATFORM_ARG} -it --tty ${DOCKER_IMAGE_NAME}:latest"
```

Reply:
I'll add it back.